src/Microsoft.AspNetCore.Server.Kestrel/Http/Frame.cs

```diff
@@ -234,14 +234,13 @@ public void Reset()
             StatusCode = 200;
             ReasonPhrase = null;
 
-            var httpConnectionFeature = this as IHttpConnectionFeature;
-            httpConnectionFeature.RemoteIpAddress = RemoteEndPoint?.Address;
-            httpConnectionFeature.RemotePort = RemoteEndPoint?.Port ?? 0;
+            RemoteIpAddress = RemoteEndPoint?.Address;
+            RemotePort = RemoteEndPoint?.Port ?? 0;
 
-            httpConnectionFeature.LocalIpAddress = LocalEndPoint?.Address;
-            httpConnectionFeature.LocalPort = LocalEndPoint?.Port ?? 0;
+            LocalIpAddress = LocalEndPoint?.Address;
+            LocalPort = LocalEndPoint?.Port ?? 0;
 
-            httpConnectionFeature.ConnectionId = ConnectionId;
```

Comment:
I now see what you're saying about it needing the interface properties to be settable. I think it only really messes with `ConnectionId` since we have the other original value in `RemoteEndPoint` and `LocalEndPoint`.

We could have a private field like `_originalConnectionId`, but I don't like it. Do you know how the perf of this iteration of the PR compares to the first iteration? I'm not really sure we need to change this code at all. 2.5% of `Reset` doesn't seem that bad.